packages/zendesk-adapter/src/change_validators/external_source_webhook.ts

```diff
@@ -0,0 +1,135 @@
+
+const createExternalSourceWebhookChangeWarning = (webhook: InstanceElement, appMessage: string): ChangeError => ({
+  elemID: webhook.elemID,
+  severity: 'Warning',
+  message: 'Changing a webhook that was installed by an external app',
+  detailedMessage: `${appMessage}If you edit this webhook (${webhook.elemID.name}), the app that created it might not work as intended.`,
+})
+
+const createDeactivationWarning = (webhook: InstanceElement, appMessage: string): ChangeError => ({
+  elemID: webhook.elemID,
+  severity: 'Warning',
+  message: 'Deactivating a webhook that was installed by an external app',
+  detailedMessage: `${appMessage}If you deactivate this webhook (${webhook.elemID.name}), the app that created it might not work as intended. You'll need to reactivate it to use it again.`,
+})
+
+const createExternalSourceChangeError = (webhook: InstanceElement): ChangeError => ({
+  elemID: webhook.elemID,
+  severity: 'Error',
+  message: 'Illegal webhook modification',
+  detailedMessage: `Cannot modify 'external_source' field of a webhook (${webhook.elemID.name})`,
+})
+
+const handleModificationChanges = ([change, appMessage]: [ModificationChange<InstanceElement>, string])
+    : ChangeError[] => {
+  const errors: ChangeError[] = []
+  const detailedChanges = detailedCompare(change.data.before, change.data.after)
+
+  // It's impossible to change external_source field in a webhook using Zendesk's api
+  if (detailedChanges.some(detailedChange => detailedChange.id.createTopLevelParentID().path[0] === 'external_source')) {
+    errors.push(createExternalSourceChangeError(change.data.after))
+  }
+
+  const wasDeactivated = change.data.before.value.status === 'active' && change.data.after.value.status === 'inactive'
+  if (wasDeactivated) {
+    errors.push(createDeactivationWarning(change.data.after, appMessage))
+  }
+
+  // Filter all the changes we already handled, if there are any other - we have a different warning for them
+  const otherChanges = detailedChanges.filter(detailedChange =>
+    ['external_source', 'status'].every(field => detailedChange.id.createTopLevelParentID().path[0] !== field))
+  if (otherChanges.length > 0) {
+    errors.push(createExternalSourceWebhookChangeWarning(change.data.after, appMessage))
+  }
+
+  return errors
+}
+
+const getAppMessageFromChange = <T extends Change<InstanceElement>>(change: T): [T, string] => {
+  const webhook = getChangeData(change)
+  const externalSource = webhook.value.external_source
+  // If the external_source contains an installation_id, we can get the app name from it
+  const appMessage = isExternalSourceWithApp(externalSource)
+    ? `This webhook was installed by the external app '${externalSource.data.installation_id.elemID.name}', `
+    : 'This webhook was installed an external app, '
+  return [change, appMessage]
+}
+
+const createAdditionError = ([change, appMessage]: [AdditionChange<InstanceElement>, string]): ChangeError => {
```

Comment:
same - why not accept a structured object? arrays are not really intended for this use case
```suggestion
const createAdditionError = ({ change, appMessage }: {
  change: AdditionChange<InstanceElement>
  appMessage: string
}): ChangeError => {
```